### PR TITLE
plugins/lsp: add templ language server

### PR DESCRIPTION
**Pull request by rsHalford**
Adds lsp support for [templ](https://templ.guide), the HTML templating language for Go.

### CODE REVIEW

File: plugins/lsp/language-servers/default.nix [hunk 1]
```diff
@@ -545,6 +545,11 @@ with lib; let
       description = "Enable taplo, for TOML";
       package = pkgs.taplo;
     }
+    {
+      name = "templ";
+      description = "Enable the templ language server for the templ HTML templating language";
+      package = pkgs.templ;
+    }
```
Comment (GaetanLepage) [inline plugins/lsp/language-servers/default.nix:552]:
```suggestion
      description = "Enable the templ language server for the templ HTML templating language";
    }
```
As the package has the same name as the LS, you don't need to specify it.

**Review by GaetanLepage — APPROVED** [review]
LGTM ! Thanks for your contribution :)

**Comment by psyos10** [comment]
Hey, new to nixos, may be my problem. I have been getting nixos-rebuild issues that point to this addition in nixvim, **`error: attribute 'templ' missing` at plugins/lsp/helpers.nix:6:15** , my config worked fine before this.

**Comment by traxys** [comment]
There could be multiple reasons for this:
- Are you using the correct nixvim branch for your environement? This means that if you are using nixpkgs/nixos/home-manager 23.11 then you need to use the 23.11 branch of nixvim
- If you are using nixpkgs/nixos/home-manager unstable they need to be up-to-date, templ was added on the 21st of december, so you need to have at least this version. Generally we only guarantee that the master of nixvim works with a very recent (not more than a week old) nixpkgs

**Comment by psyos10** [comment]
Apologies yep, the 23.11 works for me, I was initially confused by why the templ package wasn't there. My nixpkgs isn't that up to date. Thanks.

**Pull request merged**